components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java

```diff
@@ -0,0 +1,549 @@
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationManagementUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.validators.jwt.JWKSBasedJWTValidator;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.AnalyticsAttributes.SESSION_ID;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.AUTHENTICATOR_NAME;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.OIDC_BCLOGOUT_ENDPOINT_URL_PATTERN;
+
+public class LogoutProcessor extends IdentityProcessor {
+
+    private static final Log log = LogFactory.getLog(LogoutProcessor.class);
+    private SessionManagementService sessionManagementService = new SessionManagementService();
+
+    private String tenantDomain;
+
+    private static final String ENFORCE_CERTIFICATE_VALIDITY
+            = "JWTValidatorConfigs.EnforceCertificateExpiryTimeValidity";
+    private static final String DEFAULT_IDP_NAME = "default";
+    private static final String ERROR_GET_RESIDENT_IDP =
+            "Error while getting Resident Identity Provider of '%s' tenant.";
+    private static final String OIDC_IDP_ENTITY_ID = "IdPEntityId";
+    private static final String BACKCHANNEL_LOGOUT_EVENT = "http://schemas.openidnet/event/backchannel-logout";
+    private static final String BACKCHANNEL_LOGOUT_EVENT_CLAIM = "{}";
+    private static final String ENABLE_IAT_VALIDATION = "enableIatValidation";
+    private static final String IAT_VALIDITY_PERIOD = "iatValidityPeriod";
+
+    @Override
+    public IdentityResponse.IdentityResponseBuilder process(IdentityRequest identityRequest) throws FrameworkException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Request processing started by OIDC LogoutProcessor.");
+        }
+        LogoutContext logoutContext = new LogoutContext(identityRequest);
+
+        IdentityResponse.IdentityResponseBuilder identityResponseBuilder = null;
+
+        if (identityRequest instanceof LogoutRequest) {
```

Comment:
Do we need this check?
We can only reach here if this was a logout request, isn't it? (Processors canHandle() method should do this)

Reply:
fixed in 201c3de